pkg/api/v1beta1/conversion.go

```diff
@@ -444,7 +444,115 @@ func init() {
 			}
 			return nil
 		},
+		// Converts cpu and memory requirements to Resources, and vice versa.
+		func(in *newer.Container, out *Container, s conversion.Scope) error {
+			if err := s.Convert(&in.Name, &out.Name, 0); err != nil {
```

Comment:
OMG this seems error-prone. 

Could we use reflection to automatically convert all fields that are present in both in and out?

I don't want to block this PR on addressing this, but please add a TODO.

@thockin @smarterclayton Any ideas about how we could automate this?